model/artifact/artifact_file.go

```diff
@@ -1,10 +1,19 @@
 package artifact
 
 import (
+	"log"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
```

Comment:
we almost never make direct calls to s3, and always use our intermediary library pail. It's also weird to put aws code in our db interaction code. I'm not sure if there's a good resolution (and julian's been handling most pail development, but it might be nice to check.

Reply:
Are these direct calls to s3? My impression was that signing was purely on the client and never makes an actual API call. Nevertheless it is still reasonable to put it in pail, if there's an elegant way to do that.

Reply:
I felt uneasy about that too. I'll talk to Julian. Thanks!

Reply:
@bsamek @tychoish I spoke to Julian. It appears that there is no elegant way to put it in pail, because it's very specific. We can talk about it more in person if you'd like.

Reply:
Let's package this up in the "thirdparty" package in evergreen which I think is a good place for this kind of thing, at least for now.